features/steps/fastlane_build_report_steps.rb

```diff
@@ -0,0 +1,49 @@
+Given "fastlane is installed" do
+    $fastlane_installed ||= begin
+      step('I run the script "features/scripts/install-fastlane.sh" synchronously')
+      true
+    end
+  end
+  When ("I run lane {string} with app_version set to {string}") do |lane, app_version|
+    fastlane_report_build(lane, app_version)
+  end
+
+  When ("I run lane {string} with api_key also set to {string}") do |lane, api_key|
+    fastlane_report_build(lane, api_key)
+  end
+
+  When("I run lane {string} with api_key set to {string} and app_version set to {string}") do |lane, api_key, app_version|
+    fastlane_report_build(lane, api_key, app_version)
+  end
+
+  When ("I run lane {string} with api_key set to {string} and app_version set to {string} and metadata set to {string}") do |lane, api_key, app_version, metadata|
```

Comment:
With this many different combinations, I think we should rethink how we're running lanes. Is it possible to build up a configuration programmatically, like:

```
When I set fastlane api_key to "1234567890ABCDEF1234567890AAAAAA"
And I set fastlane app_version to "1.0"
And I set fastlane ios_bundle_version to "1.0.0"
And I run lane "report_build"
Then I should receive 1 request
...
```